Change "View" in Looker to "Explore" & replace rJson with RJSONIO

thekenz opened the pull request:
fixes #5 

I chose to go with `explore` over `base_view`, because I thought that `explore` would be more clear for everyone, especially users of Looker that have never seen the `base_view` parameter, and have only dealt with explores. 

Furthermore, I tested what LookR does where we alias the base view in the LookML using the `view:` parameter. 

```LookML
- explore: test
  view: order_items
```

Turns out that LookR will not return data in this instance:

```R
my_inline_query <- run_inline_query(model = "kenny_test_postgres", 
                                    explore = "order_items", 
                                    fields = c("order_items.order_id", "order_items.id"))
```

but LookR **will** return data in this instance:

```R
my_inline_query <- run_inline_query(model = "kenny_test_postgres", 
                                    explore = "test", 
                                    fields = c("order_items.order_id", "order_items.id"))
```

Thus, I changed the references in LookR to `explore`, for better clarity here.

githoov commented:
@thekenz Look great! Maybe update the README example too?

thekenz commented:
Fixes #6 

@maxcorbin and I did some digging around here, and we found out that the`fromJSON` function in the `rJson` package that we've been using doesn't have a good way to parse null values. We found that the `fromJSON` package in both [`jsonlite`](https://cran.r-project.org/web/packages/jsonlite/jsonlite.pdf) (page 8) and [`JSONIO`](https://cran.r-project.org/web/packages/RJSONIO/RJSONIO.pdf) (page 4) were a lot more robust and provided a way to handle null values a lot better here. 

Max found [this document](https://rstudio-pubs-static.s3.amazonaws.com/31702_9c22e3d1a0c44968a4a1f9656f1800ab.html) that compares rJson, rjsionio, and jsionlite. We ultimately went with jsonlite because it appeared to be more performant, and it handles matrices in a way that is similar to rJSON. 

Tested it out using this script, and everything seems to check out.

```R
library("devtools")
library("LookR")

# setting up the client and authenticating (id and secret are available on your Looker instance)
looker_setup(   id = "",
                secret = "",
                api_path = "https://learn.looker.com:19999/api/3.0"
)

# running a Look by its look_id
area_chart <- run_look(1597)
```

WOOT! Let's goooo!

thekenz commented:
Update: jsonlite actually rendered queries using `run_inline_query` in a strange way, so I went with RJSONIO instead. Went about update the dependencies, and the package imports from rJson to RJSONIO. 

Tested both using this script:

```R
# preliminaires #
library("devtools")
library("LookR")

# setting up the client and authenticating (id and secret are available on your Looker instance)
looker_setup(   id = "",
                secret = "",
                api_path = "https://learn.looker.com:19999/api/3.0"
)

# running a Look by its look_id
my_look <- run_look(1597)

# running an inline query by providing query components (note: model, view, and fields are required parameters)
my_inline_query <- run_inline_query(model = "ecommerce", 
                                    explore = "orders", 
                                    fields = c("orders.id"))
```

and the resulting dataframes seem to return very nicely.

maxcorbin commented:
@thekenz Looks awesome, good call on rolling with RJSONIO. I'm wondering if the change from `view` to `explore` for the `run_inline_query` argument is a good idea, since the underlying JSON parameter is `view`.

I definitely agree that it is confusing, I actually just had a chat a few days ago where this came up. But it feels safer to keep R function arguments named identically to the JSON parameters they represent. Maybe we can talk to Engineering and have the `view` parameter renamed in the API before we rename it here?

githoov commented:
Pulled and tested the changes, @thekenz. Looks great! Seems to fix issue #6. 

I'm still not certain whether the migration to `explore` within the function is the best way to go—I just don't know. A more passive approach is to simply keep things as `view` and update the README with a comment, pointing to the fact that `view` here just means `explore`. That way we're consistent with the API documentation. If you and @maxcorbin come to an agreement on this, I'll merge the PR.

thekenz commented:
Hey @githoov! I definitely agree with you, John, and @maxcorbin that keeping the `view` parameter inside the `run_inline_query` function is better in order to keep the functions in line with the API. I've gone ahead and changed that code back to `view` in `run_inline_query`, and added a flag in the README signifying that this parameter corresponds to the name of the explore, rather than the name of a view within an explore.

thekenz commented:
I made the requisite changes here, and I think that this pull request is ready for merge!

githoov commented on R/functions.R:
```diff
@@ -70,15 +70,15 @@ run_look <- function(look_id = NULL, format = "json"){
 
 # run inline query by building up query components
-run_inline_query <- function(model, view, fields, filters = NULL, pivots = NULL, sorts = NULL, limit = NULL){
+run_inline_query <- function(model, explore, fields, filters = NULL, pivots = NULL, sorts = NULL, limit = NULL){
```
@thekenz One more. Then I'll merge this guy!

thekenz commented:
BOOM 💥  we are good to go!
